Resolve merge conflicts from Pil Pillow conversion branch

BonfaceKilz opened the pull request:
#### Description:

This PR tries to resolve merge conflicts from: https://github.com/genenetwork/genenetwork2/pull/355.

#### Any background context you want to provide

Cherry picked all the commits from that branch. Tried a rebase and a merge but that was way too much work so I ditched that.

fredmanglis committed:
Update the colours to use Pillow's ImageColor

* wqflask/wqflask/marker_regression/display_mapping_results.py: Use Pillow's ImageColor module in place of the older Piddle colours.

fredmanglis committed:
Migrate code from Piddle to Pillow

* wqflask/wqflask/marker_regression/display_mapping_results.py: Add font files
(DisplayMappingResults): Update the code from the Piddle way of things to the
Pillow way of things, for example, replace:
- pid.drawRect(...) with im_drawer.rectangle(...)
- pid.drawString(...) with im_drawer.text(...)
- pid.drawPolygon(...) with im_drawer.polygon(...) etc.
* wqflask/utility/Plot.py: Migrate code from the older, unsupported Piddle to
the newer Pillow library.

pjotrp committed:
Add fonts for pillow

* wqflask/wqflask/marker_regression/display_mapping_results.py: replace piddle
method with PIL.ImageFont
* wqflask/wqflask/static/fonts: Add fonts in path

fredmanglis committed:
Add missing font files

* wqflask/fonts/ttf: Add ttf font files.

fredmanglis committed:
Update SNP Browser to Pillow

* wqflask/wqflask/snp_browser/snp_browser.py: Use newer, and supported Pillow in
place of obsoleted Piddle library.

fredmanglis committed:
Fix font paths

* wqflask/utility/Plot.py: Update font paths.
* wqflask/fonts/ttf: Update font files.

fredmanglis committed:
Create new utility module for drawing

* wqflask/utility/pillow_utils.py: Create a module to hold some utility
functions for drawing with Pillow. Initialise the module with a function to draw
rotated text.

fredmanglis committed:
Use draw_rotated_text()

Use the new draw_rotated_text() function to draw the text rotated as was
formerly done.

* wqflask/utility/Plot.py (plotBar): Use draw_rotated_text().
* wqflask/wqflask/marker_regression/display_mapping_results.py:
(DisplayMappingResults) Use draw_rotated_text().

fredmanglis committed:
Replace "color" with "fill" keyword argument

* wqflask/wqflask/marker_regression/display_mapping_results.py:
(DisplayMappingResults): Pillow uses 'fill' in place of 'color'.

fredmanglis committed:
Add draw_open_polygon() utility

* wqflask/utility/pillow_utils.py: New method.

fredmanglis committed:
Use new `draw_open_polygon()` utility

* wqflask/wqflask/marker_regression/display_mapping_results.py: Use the new
`draw_open_polygon()` utility to draw open polygons.

fredmanglis committed:
Give rectangles black outlines

* wqflask/wqflask/marker_regression/display_mapping_results.py: Give the
rectangles black outlines to correspond to earlier drawings.

fredmanglis committed:
Use TEXT_Y_DISPLACEMENT instead of magic numbers

* wqflask/wqflask/marker_regression/display_mapping_results.py: Use
TEXT_Y_DISPLACEMENT instead of magic numbers to help with making it clear that
the drawn text is displaced somewhat.

fredmanglis committed:
Fix placement for axes labels

* wqflask/wqflask/marker_regression/display_mapping_results.py: Fix the
placement for the axes labels.

fredmanglis committed:
Change Y Coodinate system

* wqflask/wqflask/marker_regression/display_mapping_results.py: Piddle seemed to
centre text about the given Y coordinate, while Pillow requires specification to
top-left corner of the text. This slight change means text all over the image is
displaced somewhat, and this fixes the coordinate system for some of the text.

BonfaceKilz commented:
@zsloan could you help me test this out?

zsloan commented:
I'm getting this error when mapping:
`GeneNetwork tux01:gene:2.11-rc2-HEAD-13c9fe533 http://gn2-zach.genenetwork.org/run_mapping ( 5:34PM UTC Aug 06, 2020)
 truetype() got an unexpected keyword argument 'font' (error)
     File "/export/local/home/zas1024/gn2-zach/gene/wqflask/wqflask/marker_regression/display_mapping_results.py", line 2557, in drawGraphBackground
        chrLabelFont=ImageFont.truetype(font=VERDANA_FILE,size=24*chrFontZoom)`

BonfaceKilz commented:
Ok. Let me investigate that
-- 
Bonface M. K. (https://www.bonfacemunyoki.com)
One Divine Emacs To Rule Them All
GPG key = D4F09EB110177E03C28E2FE1F5BBAE1E0392253F

BonfaceKilz commented:
@zsloan I have an idea where the problem is. That merge is not clean. What I'm trying to do is an interactive rebase to handpick commits :)

BonfaceKilz commented:
A rebase is way too complicated. I've given up on that. I think it's just easier to look at the commits, cherry pick what I can, and modify what I can. @zsloan let me have a go at this.

PS: If I can I'll add tests too.

BonfaceKilz committed:
Move wqflask/tests/api to wqflask/tests/wqflask

* wqflask/tests(api): Move all the files here to tests/wqflask. The test dir
should mirror the actual module structure

BonfaceKilz committed:
Add a basic test for test_display_marking_results.py

* wqflask/tests/wqflask/marker_regression/__init__.py: Add it
* wqflask/tests/wqflask/marker_regression/test_display_marking_results.py: Check
that PIL colors are being used instead of the deprecated Piddle colors.

BonfaceKilz commented:
@zsloan could you help me test this out? I think there's something weird I'm doing when mapping things since I'm getting(also on the testing branch too) when I tried running a mapping on *1425507_at*:


```
There is likely an issue with the genotype file associated with this group/RISet. Please contact Zach Sloan (zachary.a.sloan@gmail.com) or Arthur Centeno (acenteno@gmail.com) about the data set in question.

Try mapping using interval mapping instead; some genotype files with many columns of NAs have issues with GEMMA or R/qtl
```

My settings were:
- Mapping tool: Gemma
- MAF >= 0.05
- Use LOCO: Yes

So that ^^ means the problem's with me and some of my knowledge debts with how some things work(I'm working on paying my debts though :smile:)

I tried an interactive rebase, but that got way too complex along the way
because of some merges with in that PR. Same problem with normal merging. So
what I did is cherry pick commits that made sense to me. FWIW I edited(cc @fredmanglis)
some of the commits to stick to
https://www.gnu.org/prep/standards/html_node/Change-Logs.html.

zsloan commented:
Ah, that error is basically just to make the user not see something ugly;
you'll want to get rid of the try/except at lines 763-770 of views.py to
see the actual error -
https://github.com/genenetwork/genenetwork2/blob/testing/wqflask/wqflask/views.py

Chances are there's something going wrong in display_mapping_results.py
since that's where the module change would make a difference.

On Fri, Aug 7, 2020 at 7:57 PM BonfaceKilz <notifications@github.com> wrote:

> @zsloan <https://github.com/zsloan> could you help me test this out? I
> think there's something weird I'm doing when mapping things since I'm
> getting(also on the testing branch too) when I tried running a mapping on
> *1425507_at*:
>
> There is likely an issue with the genotype file associated with this group/RISet. Please contact Zach Sloan (zachary.a.sloan@gmail.com) or Arthur Centeno (acenteno@gmail.com) about the data set in question.
>
>
>
> Try mapping using interval mapping instead; some genotype files with many columns of NAs have issues with GEMMA or R/qtl
>
>
> My settings were:
>
>    - Mapping tool: Gemma
>    - MAF >= 0.05
>    - Use LOCO: Yes
>
> So that ^^ means the problem's with me and some of my knowledge debts with
> how some things work(I'm working on paying my debts though 😄)
>
> I tried an interactive rebase, but that got way too complex along the way
> because of some merges with in that PR. Same problem with normal merging.
> So
> what I did is cherry pick commits that made sense to me. FWIW I edited(cc
> @fredmanglis <https://github.com/fredmanglis>)
> some of the commits to stick to
> https://www.gnu.org/prep/standards/html_node/Change-Logs.html.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/genenetwork/genenetwork2/pull/417#issuecomment-670799610>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AANQJGGAMWSYGE543XCL6QLR7SPGZANCNFSM4PWRCLFA>
> .
>

BonfaceKilz commented:
Is the mapping behaving properly for you? Since I'm getting that ^^ on my latest testing branch, I'm guessing it's something that's not introduced by this PR but more like something I can look at later,

zsloan commented:
If it's the BXD trait you're looking at, it maps properly, yeah. Maybe your
branch doesn't have the genotype files (or doesn't point to them correctly
or something). That's one thing that could explain it that would be
unrelated to the PR.

On Fri, Aug 7, 2020 at 8:13 PM BonfaceKilz <notifications@github.com> wrote:

> Is the mapping behaving properly for you? Since I'm getting that ^^ on my
> latest testing branch, I'm guessing it's something that's not introduced by
> this PR but more like something I can look at later,
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/genenetwork/genenetwork2/pull/417#issuecomment-670801973>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AANQJGALN5SZKRCGKKOUCJ3R7SRE3ANCNFSM4PWRCLFA>
> .
>

BonfaceKilz commented:
Hmmm.... Yeah. You could spend more time on that too. @pjotrp or @robwwilliams  if you have time you could look at that too. If that works, we could merge that. I think this is a big step to moving to py3.8. Piddle is the only dep AFAICT that's tightly coupled to py2.

robwwilliams commented:
Bonface, point me at the specific test instance URL and will run it
tomorrow.

On Fri, Aug 7, 2020 at 8:29 PM BonfaceKilz <notifications@github.com> wrote:

> Hmmm.... Yeah. You could spend more time on that too. @pjotrp
> <https://github.com/pjotrp> or @robwwilliams
> <https://github.com/robwwilliams> if you have time you could look at that
> too. If that works, we could merge that. I think this is a big step to
> moving to py3.8. Piddle is the only dep AFAICT that's tightly coupled to
> py2.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/genenetwork/genenetwork2/pull/417#issuecomment-670803884>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC55V7EE7G3OJUQ2SSXBKELR7SS6PANCNFSM4PWRCLFA>
> .
>
-- 
Rob

Robert W. Williams, Ph.D.
Chair: Department of Genetics, Genomics and Informatics
University of Tennessee Health Science Center
71 S Manassas St, Memphis TN 38103-3310,  TSRB Room 405
Tele/Cell: 901 604 4752
EMAIL: rwilliams@uthsc.edu  or  labwilliams@gmail.com

zsloan commented:
I'll take a look at getting the mapping figure working with Pillow. I feel
like I looked at this a while back and there was something that made it
difficult/tedious (like something having to be drawn differently such that
the actual logic had to be changed), but maybe I'll notice something when
looking at it again.

On Fri, Aug 7, 2020 at 8:29 PM BonfaceKilz <notifications@github.com> wrote:

> Hmmm.... Yeah. You could spend more time on that too. @pjotrp
> <https://github.com/pjotrp> or @robwwilliams
> <https://github.com/robwwilliams> if you have time you could look at that
> too. If that works, we could merge that. I think this is a big step to
> moving to py3.8. Piddle is the only dep AFAICT that's tightly coupled to
> py2.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/genenetwork/genenetwork2/pull/417#issuecomment-670803884>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AANQJGEMR2WHSTWNBDX5253R7SS6PANCNFSM4PWRCLFA>
> .
>

pjotrp commented:
@zsloan @fredmanglis did all the hard work already. We just need to merge his stuff and that is what @BonfaceKilz is doing. The new drawing should be done. We just need to test to make sure there are no major differences.  I propose to put @chfi browser on the front anyway and have this as a second tab.

BonfaceKilz commented:
@robwwilliams here's what I used: http://gn2-test2.genenetwork.org/show_trait?trait_id=1451401_a_at&dataset=HC_M2_0606_P to test the mapping. I also added the python2-pillow dep here: http://git.genenetwork.org/guix-bioinformatics/guix-bioinformatics/commit/b0c38d151324e37448ade758cc48d02d89f94b60 (so @zsloan you should update your guix profile in order for PIL to work).Things seem to be working now after adding c1b1c91 above.

Also, you could use gn2-test2.genenetwork.org to test the mappings. I'll create another branch and harvest the piddle->pillow commits and run the py3 migration and fix anything that comes up.

@zsloan or @pjotrp if there are no issues, we shoud get this merged(?).

BonfaceKilz commented:
> I'll take a look at getting the mapping figure working with Pillow. I feel like I looked at this a while back and there was something that made it difficult/tedious (like something having to be drawn differently such that the actual logic had to be changed), but maybe I'll notice something when looking at it again.
> […](#)
> On Fri, Aug 7, 2020 at 8:29 PM BonfaceKilz ***@***.***> wrote: Hmmm.... Yeah. You could spend more time on that too. @pjotrp <https://github.com/pjotrp> or @robwwilliams <https://github.com/robwwilliams> if you have time you could look at that too. If that works, we could merge that. I think this is a big step to moving to py3.8. Piddle is the only dep AFAICT that's tightly coupled to py2. — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub <[#417 (comment)](https://github.com/genenetwork/genenetwork2/pull/417#issuecomment-670803884)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AANQJGEMR2WHSTWNBDX5253R7SS6PANCNFSM4PWRCLFA> .

I second @pjotrp there. All the heavy lifting's already been done. Now we should just check that things run as expected :)

BonfaceKilz commented:
> I'm getting this error when mapping:
> `GeneNetwork tux01:gene:2.11-rc2-HEAD-13c9fe533 http://gn2-zach.genenetwork.org/run_mapping ( 5:34PM UTC Aug 06, 2020) truetype() got an unexpected keyword argument 'font' (error) File "/export/local/home/zas1024/gn2-zach/gene/wqflask/wqflask/marker_regression/display_mapping_results.py", line 2557, in drawGraphBackground chrLabelFont=ImageFont.truetype(font=VERDANA_FILE,size=24*chrFontZoom)`

@zsloan Also resolved this today, so you could try testing again. Things should work :)

zsloan commented:
I don't want to change/add a guix profile without Pjotr's say, though I
think he mentioned sometime recently that he'd explain how to create/add to
a profile for testing purposes.

On Mon, Aug 10, 2020 at 12:08 PM BonfaceKilz <notifications@github.com>
wrote:

> I'm getting this error when mapping:
> GeneNetwork tux01:gene:2.11-rc2-HEAD-13c9fe533
> http://gn2-zach.genenetwork.org/run_mapping ( 5:34PM UTC Aug 06, 2020)
> truetype() got an unexpected keyword argument 'font' (error) File
> "/export/local/home/zas1024/gn2-zach/gene/wqflask/wqflask/marker_regression/display_mapping_results.py",
> line 2557, in drawGraphBackground
> chrLabelFont=ImageFont.truetype(font=VERDANA_FILE,size=24*chrFontZoom)
>
> @zsloan <https://github.com/zsloan> Also resolved this today, so you
> could try testing again. Things should work :)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/genenetwork/genenetwork2/pull/417#issuecomment-671476738>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AANQJGBB2MTVCYG7KAU4ZUTSAASQTANCNFSM4PWRCLFA>
> .
>

pjotrp commented:
I'll update the profile tomorrow. Ping me if I forget. Yes, @zsloan, Guix profiles are a bit different. Leave it to Bonface, Efraim and me. Currently I am the only one of the team who can update tux01. We'll change that later this year. We can teach you how to update Guix profiles if you are interested.

zsloan commented:
Yeah, that'd be useful to learn and would save some time in the future.
Just let me know at some point when it's convenient to explain it.

On Mon, Aug 10, 2020 at 1:13 PM Pjotr Prins <notifications@github.com>
wrote:

> I'll update the profile tomorrow. Ping me if I forget. Yes, @zsloan
> <https://github.com/zsloan>, Guix profiles are a bit different. Leave it
> to Bonface, Efraim and me. Currently I am the only one of the team who can
> update tux01. We'll change that later this year. We can teach you how to
> update Guix profiles if you are interested.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/genenetwork/genenetwork2/pull/417#issuecomment-671508371>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AANQJGAJGX2XXWYDNO2IK6TSAA2FHANCNFSM4PWRCLFA>
> .
>

robwwilliams commented:
@bonfaceKilz @zsloan

All looks good wrt mapping. Ran GEMMA and Haley-Knott Regression, and R/qtl1

All work great.

[image: image.png]
I do see a very minor glitch on the Chr label
when viewing a single chromosome.
[image: Screen Shot 2020-08-10 at 1.31.23 PM.png]

@zslaon

0. Hitting the horizontal link bars kills the page—these links should ALL
generate a NEW page. Only way to get back involves a complete recompute.

Please avoid one-way streets! Users like me have a fit.

1. Also the green horizontal click bar still has green font in a green
field—ergo, unreadable.
2. Also note link to USCS goes to mm9 and should point to mm10 (and
generated some ugly errors)

[image: Screen Shot 2020-08-10 at 1.35.19 PM.png]


On Mon, Aug 10, 2020 at 11:44 AM BonfaceKilz <notifications@github.com>
wrote:

> @robwwilliams <https://github.com/robwwilliams> here's what I used:
> http://gn2-test2.genenetwork.org/show_trait?trait_id=1451401_a_at&dataset=HC_M2_0606_P
> to test the mapping. I also added the python2-pillow dep here:
> http://git.genenetwork.org/guix-bioinformatics/guix-bioinformatics/commit/b0c38d151324e37448ade758cc48d02d89f94b60(so
> @zsloan <https://github.com/zsloan> you should update your guix profile
> in order for PIL to work).Things seem to be working now after adding
> c1b1c91
> <https://github.com/genenetwork/genenetwork2/commit/c1b1c914b10781715bb6a2e7fc061e6e506fb92f>
> above.
>
> Also, you could use gn2-test2.genenetwork.org to test the mappings. I'll
> create another branch and harvest the piddle->pillow commits and run the
> py3 migration and fix anything that comes up.
>
> @zsloan <https://github.com/zsloan> or @pjotrp <https://github.com/pjotrp>
> if there are no issues, we shoud get this merged(?).
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/genenetwork/genenetwork2/pull/417#issuecomment-671464542>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC55V7CW44J74YIM5CBK2ALSAAPVTANCNFSM4PWRCLFA>
> .
>


-- 
Rob

Robert W. Williams, Ph.D.
Chair: Department of Genetics, Genomics and Informatics
University of Tennessee Health Science Center
71 S Manassas St, Memphis TN 38103-3310,  TSRB Room 405
Tele/Cell: 901 604 4752
EMAIL: rwilliams@uthsc.edu  or  labwilliams@gmail.com

BonfaceKilz commented:
robwwilliams <notifications@github.com> writes:

> @bonfaceKilz @zsloan
>
> All looks good wrt mapping. Ran GEMMA and Haley-Knott Regression, and R/qtl1
>
> All work great.
>
> [image: image.png]
> I do see a very minor glitch on the Chr label
> when viewing a single chromosome.
> [image: Screen Shot 2020-08-10 at 1.31.23 PM.png]
>

Thanks for the feedback @robwwilliams! I don't think the screenshot got
attached(I can't see it on e-mail or on GH). Could you re-send the
images?

-- 
Bonface M. K. (https://www.bonfacemunyoki.com)
One Divine Emacs To Rule Them All
GPG key = D4F09EB110177E03C28E2FE1F5BBAE1E0392253F

zsloan commented:
I think the screenshots may not have worked because it was a reply on
github?

On Mon, Aug 10, 2020 at 2:03 PM BonfaceKilz <notifications@github.com>
wrote:

> robwwilliams <notifications@github.com> writes:
>
> > @bonfaceKilz @zsloan
> >
> > All looks good wrt mapping. Ran GEMMA and Haley-Knott Regression, and
> R/qtl1
> >
> > All work great.
> >
> > [image: image.png]
> > I do see a very minor glitch on the Chr label
> > when viewing a single chromosome.
> > [image: Screen Shot 2020-08-10 at 1.31.23 PM.png]
> >
>
> Thanks for the feedback @robwwilliams! I don't think the screenshot got
> attached(I can't see it on e-mail or on GH). Could you re-send the
> images?
>
> --
> Bonface M. K. (https://www.bonfacemunyoki.com)
> One Divine Emacs To Rule Them All
> GPG key = D4F09EB110177E03C28E2FE1F5BBAE1E0392253F
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/genenetwork/genenetwork2/pull/417#issuecomment-671532250>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AANQJGEJICFDBCYJNNAH53DSAA76ZANCNFSM4PWRCLFA>
> .
>

pjotrp commented:
Let's merge into testing tomorrow. Looking good!

zsloan commented:
Ah, looking at Rob's screenshots there's also an issue with the manhattan
plot points going below the X axis line.

My guess is that something (tied to the newer library) about the way it's
drawing both the Chromosome labels and manhattan plot points (which I
believe are actually characters, so they're drawn the same way as the Chr
labels) is causing their position to be altered a bit.

On Mon, Aug 10, 2020 at 2:58 PM Pjotr Prins <notifications@github.com>
wrote:

> Let's merge into testing tomorrow. Looking good!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/genenetwork/genenetwork2/pull/417#issuecomment-671557794>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AANQJGFZVH6B4RK4FZGQ2KLSABGP3ANCNFSM4PWRCLFA>
> .
>

zsloan commented:
After using the new guix profile mapping works, with the only significant
issues being the location of the Chr text when zoomed in on a chromosome
and manhattan plot points going below the X-axis line.

I'll see if there's an easy way to fix those.

On Mon, Aug 10, 2020 at 3:56 PM Zachary Sloan <zachary.a.sloan@gmail.com>
wrote:

> Ah, looking at Rob's screenshots there's also an issue with the manhattan
> plot points going below the X axis line.
>
> My guess is that something (tied to the newer library) about the way it's
> drawing both the Chromosome labels and manhattan plot points (which I
> believe are actually characters, so they're drawn the same way as the Chr
> labels) is causing their position to be altered a bit.
>
> On Mon, Aug 10, 2020 at 2:58 PM Pjotr Prins <notifications@github.com>
> wrote:
>
>> Let's merge into testing tomorrow. Looking good!
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/genenetwork/genenetwork2/pull/417#issuecomment-671557794>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AANQJGFZVH6B4RK4FZGQ2KLSABGP3ANCNFSM4PWRCLFA>
>> .
>>
>

pjotrp commented:
Wonderful conclusion to a lot of work by @fredmanglis, @BonfaceKilz and @zsloan . This is one of the last real blockers for GN2 Python3 migration. After @chfi rewriting qtlreaper we only have some minor things to fix. Excellent progress!!

zsloan commented:
I just made a lot of changes to the positioning of text to address those
issues.

Rob, can you test on my branch whenever you get the chance? Make sure to
try a variety of options, since some of the issues I found were related to
doing mapping in centimorgans.

Also, while looking into this I found a couple separate bugs where 1.
remapping after doing a map in centimorgans reverts the units to megabases
and 2. selecting cM for QTL Reaper and then doing mapping in GEMMA causes
the GEMMA map to bein cM (it's probably taking that dropdown as input).

On Wed, Aug 12, 2020 at 2:52 PM Pjotr Prins <notifications@github.com>
wrote:

> Wonderful conclusion to a lot of work by @fredmanglis
> <https://github.com/fredmanglis>, @BonfaceKilz
> <https://github.com/BonfaceKilz> and @zsloan <https://github.com/zsloan>
> . This is one of the last real blockers for GN2 Python3 migration. After
> @chfi <https://github.com/chfi> rewriting qtlreaper we only have some
> minor things to fix. Excellent progress!!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/genenetwork/genenetwork2/pull/417#issuecomment-673077183>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AANQJGHQOGK2SE7H6PL5QMDSALXJTANCNFSM4PWRCLFA>
> .
>

zsloan committed:
Fix positioning for many instances of text in the mapping figure

* wqflask/wqflask/marker_regression/display_mapping_results.py: Many text
instances had their positions changed by the switch from piddle to PIL. Change
various Y offsets and in some cases changing the logic of the way text is
positioned related to its length and font size.
* wqflask/wqflask/static/fonts/verdanab.ttf: Add bold Verdana, since there needs
to be a separate font file for bold text.

robwwilliams commented:
Zach:

Trait that I  tested

http://gn2-zach.genenetwork.org/show_trait?trait_id=18435&dataset=BXDPublish

[image: Itvl_gh9dOQQz.png]

No serious issues here, just the usual *y*-axis issue going up to value of
6.

Other minor issues—also old news.
1. Row shading below is wrong
2.  -log(p) get split on two lines which looks odd
3. Chr and Mb should be combined for sorting convenience
4. Clicking on Add function kills the map and not way to get back
5. Button styles need to be made uniform

[image: Screen Shot 2020-08-12 at 4.30.27 PM.png]

Single chromosome view looks perfect to me.

[image: Itvl_AwvvL8bN.png]
with small exceptions
1. Green font on green background for PhenoGen link
2. UCSC Genome Browser  link is mm9 and also kills map


On Wed, Aug 12, 2020 at 3:44 PM zsloan <notifications@github.com> wrote:

> I just made a lot of changes to the positioning of text to address those
> issues.
>
> Rob, can you test on my branch whenever you get the chance? Make sure to
> try a variety of options, since some of the issues I found were related to
> doing mapping in centimorgans.
>
> Also, while looking into this I found a couple separate bugs where 1.
> remapping after doing a map in centimorgans reverts the units to megabases
> and 2. selecting cM for QTL Reaper and then doing mapping in GEMMA causes
> the GEMMA map to bein cM (it's probably taking that dropdown as input).
>
> On Wed, Aug 12, 2020 at 2:52 PM Pjotr Prins <notifications@github.com>
> wrote:
>
> > Wonderful conclusion to a lot of work by @fredmanglis
> > <https://github.com/fredmanglis>, @BonfaceKilz
> > <https://github.com/BonfaceKilz> and @zsloan <https://github.com/zsloan>
> > . This is one of the last real blockers for GN2 Python3 migration. After
> > @chfi <https://github.com/chfi> rewriting qtlreaper we only have some
> > minor things to fix. Excellent progress!!
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/genenetwork/genenetwork2/pull/417#issuecomment-673077183
> >,
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AANQJGHQOGK2SE7H6PL5QMDSALXJTANCNFSM4PWRCLFA
> >
> > .
> >
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/genenetwork/genenetwork2/pull/417#issuecomment-673101598>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC55V7CO5L2IVST3FNN4HPTSAL5MNANCNFSM4PWRCLFA>
> .
>


-- 
Rob

Robert W. Williams, Ph.D.
Chair: Department of Genetics, Genomics and Informatics
University of Tennessee Health Science Center
71 S Manassas St, Memphis TN 38103-3310,  TSRB Room 405
Tele/Cell: 901 604 4752
EMAIL: rwilliams@uthsc.edu  or  labwilliams@gmail.com

zsloan commented:
The y axis issue is still fixed; it only shows up currently because the
pull request is using older code that doesn't include the fix. After
merging it the issue will be gone again.

Other things are unrelated to the actual PIL-pillow conversion (though I'll
still fix them of course); the main thing I was wanting to check is that
the actual map image looked okay, because the conversion had caused some
positions to be altered.

On Wed, Aug 12, 2020 at 4:37 PM robwwilliams <notifications@github.com>
wrote:

> Zach:
>
> Trait that I tested
>
>
> http://gn2-zach.genenetwork.org/show_trait?trait_id=18435&dataset=BXDPublish
>
> [image: Itvl_gh9dOQQz.png]
>
> No serious issues here, just the usual *y*-axis issue going up to value of
> 6.
>
> Other minor issues—also old news.
> 1. Row shading below is wrong
> 2. -log(p) get split on two lines which looks odd
> 3. Chr and Mb should be combined for sorting convenience
> 4. Clicking on Add function kills the map and not way to get back
> 5. Button styles need to be made uniform
>
> [image: Screen Shot 2020-08-12 at 4.30.27 PM.png]
>
> Single chromosome view looks perfect to me.
>
> [image: Itvl_AwvvL8bN.png]
> with small exceptions
> 1. Green font on green background for PhenoGen link
> 2. UCSC Genome Browser link is mm9 and also kills map
>
>
> On Wed, Aug 12, 2020 at 3:44 PM zsloan <notifications@github.com> wrote:
>
> > I just made a lot of changes to the positioning of text to address those
> > issues.
> >
> > Rob, can you test on my branch whenever you get the chance? Make sure to
> > try a variety of options, since some of the issues I found were related
> to
> > doing mapping in centimorgans.
> >
> > Also, while looking into this I found a couple separate bugs where 1.
> > remapping after doing a map in centimorgans reverts the units to
> megabases
> > and 2. selecting cM for QTL Reaper and then doing mapping in GEMMA causes
> > the GEMMA map to bein cM (it's probably taking that dropdown as input).
> >
> > On Wed, Aug 12, 2020 at 2:52 PM Pjotr Prins <notifications@github.com>
> > wrote:
> >
> > > Wonderful conclusion to a lot of work by @fredmanglis
> > > <https://github.com/fredmanglis>, @BonfaceKilz
> > > <https://github.com/BonfaceKilz> and @zsloan <
> https://github.com/zsloan>
> > > . This is one of the last real blockers for GN2 Python3 migration.
> After
> > > @chfi <https://github.com/chfi> rewriting qtlreaper we only have some
> > > minor things to fix. Excellent progress!!
> > >
> > > —
> > > You are receiving this because you were mentioned.
> > > Reply to this email directly, view it on GitHub
> > > <
> >
> https://github.com/genenetwork/genenetwork2/pull/417#issuecomment-673077183
> > >,
> > > or unsubscribe
> > > <
> >
> https://github.com/notifications/unsubscribe-auth/AANQJGHQOGK2SE7H6PL5QMDSALXJTANCNFSM4PWRCLFA
> > >
> > > .
> > >
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/genenetwork/genenetwork2/pull/417#issuecomment-673101598
> >,
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AC55V7CO5L2IVST3FNN4HPTSAL5MNANCNFSM4PWRCLFA
> >
> > .
> >
>
>
> --
> Rob
>
> Robert W. Williams, Ph.D.
> Chair: Department of Genetics, Genomics and Informatics
> University of Tennessee Health Science Center
> 71 S Manassas St, Memphis TN 38103-3310, TSRB Room 405
> Tele/Cell: 901 604 4752
> EMAIL: rwilliams@uthsc.edu or labwilliams@gmail.com
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/genenetwork/genenetwork2/pull/417#issuecomment-673123402>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AANQJGHE4RKIO2RQGDUFOPDSAMDQLANCNFSM4PWRCLFA>
> .
>

BonfaceKilz commented:
@zsloan I think this PR is good to go after adding d1570191 which addresses some
of the issues raised above. We could get this merged. Wdyt?

Btw, thanks for sending the patch :)

zsloan commented:
I think it should be good to go. I'll go ahead and merge and pull on my branch first just to make sure.

zsloan commented:
After the merge a change to the text on the map was lost (I'm pretty sure this is because the way the text is drawn was changed). So I'll fix that to draw the same text before putting on production.

robwwilliams commented:
great progress

On Mon, Aug 17, 2020 at 5:21 PM zsloan <notifications@github.com> wrote:

> After the merge a change to the text on the map was lost (I'm pretty sure
> this is because the way the text is drawn was changed). So I'll fix that to
> draw the same text before putting on production.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/genenetwork/genenetwork2/pull/417#issuecomment-675144024>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC55V7BOUUNYQV774CKW4YTSBGUMXANCNFSM4PWRCLFA>
> .
>


-- 
Rob

Robert W. Williams, Ph.D.
Chair: Department of Genetics, Genomics and Informatics
University of Tennessee Health Science Center
71 S Manassas St, Memphis TN 38103-3310,  TSRB Room 405
Tele/Cell: 901 604 4752
EMAIL: rwilliams@uthsc.edu  or  labwilliams@gmail.com
